internal/vcs/git/files.go

```diff
@@ -0,0 +1,191 @@
+package git
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"strconv"
+	"strings"
+
+	"github.com/simplesurance/baur/v3/internal/exec"
+)
+
+const (
+	// source of the descriptions: git-ls-files manpage (Git 2.43.0)
+
+	// H tracked file that is not either unmerged or skip-worktree
+	ObjectStatusCached byte = 'H'
+	// S tracked file that is skip-worktree
+	ObjectStatusSkipWorktree byte = 'S'
+	// M tracked file that is unmerged
+	ObjectStatusUnmerged byte = 'M'
+	// R tracked file with unstaged removal/deletion
+	ObjectStatusRemoved byte = 'R'
+	// C tracked file with unstaged modification/change
+	ObjectStatusChanged byte = 'C'
+	// K untracked paths which are part of file/directory conflicts which
+	// prevent checking out tracked files
+	ObjectStatusToBeKilled byte = 'K'
+	// ? untracked file
+	ObjectStatusUntracked byte = '?'
+	// U file with resolve-undo information
+	ObjectStatusWithResolveUndoInfo byte = 'U'
+)
+
+const (
+	ObjectTypeSymlink = 0120000
+	ObjectTypeFile    = 0100000
+)
+
+type Object struct {
+	Status byte
+	Mode   uint32
+	// Name is the hash of the object
+	Name    string
+	RelPath string
+}
+
+func (o *Object) String() string {
+	return fmt.Sprintf("status: %c mode: %o name: %s path: %s", o.Status, o.Mode, o.Name, o.RelPath)
+}
+
+func (o *Object) IsSymlink() bool {
+	return o.Mode&ObjectTypeSymlink == ObjectTypeSymlink
+}
+
+func (o *Object) IsFile() bool {
+	return o.Mode&ObjectTypeFile == ObjectTypeFile
+}
+
+func scanNullTerminatedLines(data []byte, atEOF bool) (advance int, token []byte, err error) {
+	if atEOF && len(data) == 0 {
+		return 0, nil, nil
+	}
+	if i := bytes.IndexByte(data, '\x00'); i >= 0 {
+		return i + 1, data[0:i], nil
+	}
+	if atEOF {
+		return len(data), data, nil
+	}
+	return 0, nil, nil
+}
+
+// LsFiles retrieves information about files in the index and working tree.
+// The information is sent to ch.
+// For a path multiple objects might be sent, if the status is in index and working tree differs.
+// The function closes ch when it terminates.
+func LsFiles(ctx context.Context, dir string, skipUntracked bool, ch chan<- *Object) error {
+	defer close(ch)
+	// TODO: test with paths containing spaces
```

Comment:
shouldn't it "just work" ?

Reply:
Thanks, I forgot about that.

Yes, if the output is parsed correctly. If -z is not passed to git ls-files whitespaces and other special chars will also appear quoted.
I changed an existing  testcase to use a filename with whitespaces and utf8 chars to test it.